src/domFacade/DomFacade.ts

```diff
@@ -1,135 +1,485 @@
+import {
```

Comment:
Can we try to reduce the amount of casts in this file? There currently seem to 48 casts happening here.

I got the feeling that we either should take another look at the types we are using or rewrite it a bit so we don't have cast this often.